stan/math/opencl/matrix_cl.hpp

```diff
@@ -227,10 +222,12 @@ class matrix_cl<T, require_arithmetic_t<T>> {
    * Constructor for the matrix_cl that
    * creates a copy of the Eigen matrix on the OpenCL device.
    *
-   * @param A the Eigen matrix
+   * @param A the vector of  Eigen matrices
    *
    * @throw <code>std::invalid_argument</code> if the
    * matrices do not have matching dimensions
+   * @throw <code>std::system_error</code> if the memory on the device could not
+   * be allocated
```

Comment:
I think you lost part of this comment in a merge. Same with the changes in lines 377 and 391 of the original refactor, but I can't comment there since there are no changes there in your PR.